Mathlib/LinearAlgebra/QuadraticForm/Basis.lean

```diff
@@ -18,7 +19,50 @@ open LinearMap (BilinMap)
 
 namespace QuadraticMap
 
-variable {ι R M N} [LinearOrder ι]
+section
+
+variable {ι R M N}
+
+variable [CommRing R] [AddCommGroup M] [AddCommGroup N] [Module R M] [Module R N] [DecidableEq ι]
+
+-- c.f. `_root_.map_finsupp_sum`
+open Finsupp in
+theorem map_finsuppSum (Q : QuadraticMap R M N) (f : ι →₀ R) (g : ι → R → M) :
+    Q (f.sum g) =
+      f.sum (fun i r => Q (g i r)) +
+      ∑ p ∈ f.support.sym2 with ¬ p.IsDiag,
+        p.lift ⟨fun i j => polar Q (g i (f i)) (g j (f j)), fun _ _ => polar_comm _ _ _⟩ :=
+  QuadraticMap.map_sum _ _ _
```

Comment:
Could you add a copy of all three lemmas proved with `map_sum'` as well, without the `IsDiag` condition?

Reply:
I assume something like:
```
lemma testsum (f : ι →  ι → N) (hf : ∀ (a₁ a₂ : ι), f a₁ a₂ = f a₂ a₁) (s : Finset ι) :
    ∑ p ∈ s.sym2 with p.IsDiag,p.lift ⟨f, hf⟩ = ∑ i ∈ s, f i i := sorry
```
must be true, but I can't see how to prove it?